pkg/sysinfo/hostname/hostname_lin_win.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+// +build linux windows
+
+package hostname
+
+import (
+	"errors"
+	"github.com/newrelic/infrastructure-agent/pkg/trace"
+	"net"
+	"strings"
+)
+
+// Looks up for the Fully Qualified Domain Name.
+// `localhost` should not be returned as FQDN, but until deciding how it affects we will maintain this version
```

Comment:
Rather than duplicating the code, could we make this check in runtime?

We could, for example, make the function accept a "blacklist" of hostnames, and keep two lists for the different OSes. Then we could use one or another with `runtime.GOOS`.